values/encoding/int_encode.go

```diff
@@ -42,39 +41,45 @@ type intEncoder struct {
 // NewIntEncoder creates a new int encoder.
 func NewIntEncoder() IntEncoder {
 	return &intEncoder{
-		// Make buf at least big enough to hold Uint64 values.
-		buf: make([]byte, uint64SizeBytes),
+		// Make buf at least big enough to hold varint64 values.
+		buf: make([]byte, binary.MaxVarintLen64),
 		// Create a bitWriter w/ a nil write buffer that will be re-used for every Encode call.
 		bitWriter: bitstream.NewWriter(nil),
 	}
 }
 
 // Encode encodes a collection of ints and writes the encoded bytes to the writer.
-func (enc *intEncoder) Encode(intVals values.IntValues, writer io.Writer) error {
+func (enc *intEncoder) Encode(
+	intVals values.IntValues,
+	writer io.Writer,
+) error {
 	// Reset internal state at the beginning of every `Encode` call.
 	enc.reset(writer)
 
 	// Determine whether we want to do dictionary compression.
 	var (
-		valueMeta             = intVals.Metadata()
-		maxCardinalityAllowed = int(math.Ceil(1.0 * float64(valueMeta.Size) * intDictEncodingMaxCardinalityPercent))
-		dictionary            = make(map[int]int, maxCardinalityAllowed)
-		idx                   int
+		valueMeta                                     = intVals.Metadata()
+		maxCardinalityAllowed                         = int(math.Ceil(1.0 * float64(valueMeta.Size) * intDictEncodingMaxCardinalityPercent))
+		dictionary                                    = make(map[int]int, maxCardinalityAllowed)
+		idx                                           int
+		estimatedNumOfBytesRequiredForRawSizeEncoding int
```

Comment:
This is a bit too verbose. Perhaps `numRawEncodingBytes`?

Reply:
If you use `varintEncode` above, then use `numVarintEncodingBytes` here to be consistent.